src/Components/itemDetail.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import '../App.css';
+import { useParams } from 'react-router-dom';
+import { getLocalToken } from '../lib/token.js';
+import { FirestoreDocument } from 'react-firestore';
+import { Link } from 'react-router-dom';
+
+function ItemDetailPage() {
+  let { docId } = useParams();
+  // date formater
+
+  // calculate days until next purchase
+  const getFormattedDate = date => {
+    const seconds = date.seconds;
+    const year = Math.floor(seconds / 31536000);
+    const nummonths = Math.floor((seconds % 31536000) / 2628000);
+    const numdays = Math.floor(((seconds % 31536000) % 2628000) / 86400);
```

Comment:
One thing that you can consider moving forward, is to create constants for all these numbers. Reasons for this:

1. It's pretty easy to add a wrong number in these 3 lines, for example, I could type `31536000` on one line, and then type `31530600` in the next one.
2. It will give more meaning to the code, `86400` could be `SECONDS_IN_24_HOURS` and so forth.